README.md

```diff
@@ -12,24 +12,67 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
 
 ## Running
 
+### Running with the SSH Build Agents plugin
+
 To run a Docker container
 
 ```bash
-docker run jenkins/ssh-agent "<public key>"
+docker run -d --rm --name=agent -p 4444:22 -e "JENKINS_AGENT_SSH_PUBKEY=<public_key>" jenkins/ssh-agent
 ```
 
-You'll then be able to connect this agent using the [SSH Build Agents plugin](https://plugins.jenkins.io/ssh-slaves) as "jenkins" with the matching private key.
+ - `-d`: To start a container in detached mode, use the `-d` option. Containers started in detached mode exit when the root process used to run the container exits, unless you also specify the --rm option.
+ - `--rm`: If you use -d with --rm, the container is removed when it exits or when the daemon exits, whichever happens first.
+ - `--name`: Assigns a name to the container
+ - `-p`: Publishes a container's port(s) to the host. Here, we publish the container SSH port 22 to the host port 4444.
+
+Please note none of these options are mandatory, they are just examples.
+
+You will then be able to connect this agent using the [SSH Build Agents plugin](https://plugins.jenkins.io/ssh-slaves) as "jenkins" with the matching private key.
+
+When using the Linux image, you have to set the value of the `Remote root directory` to `/home/jenkins/agent` in the agent configuration UI.
+
+![Remote root directory with a Linux agent](docs/ssh-plugin-remote-root-directory-linux.png "Remote root directory with a Linux agent")
+
+When using the Windows image, you have to set the value of the `Remote root directory` to `C:/Users/jenkins/Work` in the agent configuration UI.
+
+![Remote root directory with a Windows agent](docs/ssh-plugin-remote-root-directory-windows.png "Remote root directory with a Windows agent")
+
+If you intend to use another directory than `/home/jenkins/agent` under Linux or `C:/Users/jenkins/Work` under Windows, don't forget to add it as a data volume.
+
+```bash
+docker run -v my-local-dir:path-on-the-jenkins-ssh-agent:rw jenkins/ssh-agent "<public key>"
```

Comment:
```suggestion
docker run -v docker-volume-for-jenkins-ssh-agent:/home/jenkins/agent:rw jenkins/ssh-agent "<public key>"
```

- Prefer recommending a data volume by default 
- Is this snippet to show where to mount with the default mount path, or is it about specifying another one?

Reply:
Would the [${AGENT_WORKDIR}](url) variable defined in the [`Dockerfile`](https://github.com/gounthar/docker-ssh-agent/blob/volume-one-liner/11/bullseye/Dockerfile#L62) still be visible?
I could totally be wrong though.... 🤔 
That would give something like:
`docker run -v docker-volume-for-jenkins-ssh-agent:${AGENT_WORKDIR}:rw jenkins/ssh-agent "<public key>"

Sorry, I'm muddying the water there...
My initial goal was to describe a specific new path, and not how to use the default one. 😊 
So please forget what I wrote a few lines above.
`

Reply:
> Would the [${AGENT_WORKDIR}](https://github.com/jenkinsci/docker-ssh-agent/pull/url) variable defined in the [Dockerfile](https://github.com/gounthar/docker-ssh-agent/blob/volume-one-liner/11/bullseye/Dockerfile#L62) still be visible?
> That would give something like:
> `docker run -v docker-volume-for-jenkins-ssh-agent:${AGENT_WORKDIR}:rw jenkins/ssh-agent ""

Only *inside* the container itself. In your example here, the variable would be resolved by the shell where the command is run.

While `docker run jenkins/ssh-agent env` would show you that `AGENT_WORKDIR` is in the container's environment.

>  My initial goal was to describe a specific new path, and not how to use the default one. 😊

Make sense. WDYT of something like that (adding the intent before the code snippet + a screenshot of the Jenkins UI with the correct value after the snippet):

````
For instance if you want to use `/workspace` as the agent working directory in the container mounted from a Docker volume named `docker-volume-for-jenkins-ssh-agent`

```bash
docker run --volume docker-volume-for-jenkins-ssh-agent:/workspace:rw jenkins/ssh-agent "<public key>"
```

and configure the "Remote Working Directory" to `/workspace`:

<screenshot here>
````